Refined backup and recovery sections (#76)
cncf/sig-app-delivery: operator-wg/whitepaper/03_Capabilities.md

> An operator should report the version of the resources and their health status during the process. If there was an error, the version reported should be the version that is actually been used.

Reviewer: Do you mean the current version used? 
```suggestion
An operator should report the version of the resources and their health status during the process. If there was an error, the version reported should be the version that is currently been used.
```

#### Backup

> This capability ensures that an operator is able to create consistent backups. This backup should be done in a way that the user of the operator can be certain that it can be restored if data is lost or compromised. Furthermore, the status information provided, should give insights when the backup ran the last time and where it is located.

Reviewer 2: ```suggestion
This capability is for operators that manage data and ensure that the operator is able to create consistent backups. This backup should be done in a way that the user of the operator can be certain that it can be restored if data is lost or compromised. Furthermore, the status information provided, should give insights when the backup ran the last time and where it is located.
```

> This capability is for operators that manage data and ensure that the operator is able to create consistent backups. This backup should be done in a way that the user of the operator can be certain that it can be restored if data is lost or compromised. Furthermore, the status information provided, should give insights when the backup ran the last time and where it is located.

Reviewer: ```suggestion
This capability is for operators that manage data and ensure that the operator is able to create consistent backups. This backup should be done in a way that the user of the operator can be certain that it can be restored if data is lost or compromised. Furthermore, the status information provided should give insights about when the backup last ran and where it is located.
```

> ![Example Backup Process](plantuml/backup-sequence.png)

Reviewer 2: Maybe add in the illustration the external storage, for clarity

Author: As there is a lot of ways to back up the data, i think the current drawing is the most neutral one (but i get your point) ...

> Above illustration shows how such an process could look like. At first the backup gets triggered either by a human or another trigger (e.g. time-trigger). The operator instructs its watched resource (application) to set up a consistent state (like a consistent snapshot). Afterwards, the data of the application gets backed up using appropriate tools to an external storage. This could either be a one step process (backup directly to external storage) or in multiple steps, like writing to a persistent volume at first and to the external storage afterwards. The external storage might be a NFS/CIFS share (or any other network file system) on-premises, but also an object store/bucket on a cloud provider infrastructure. Either if the backup failed or succeeded, the state (of the backup) including the backed up application version and the location of the backup might be written to the state section of the custom resource.

Reviewer 2: Maybe add in the illustration the external storage, for clarity

> Above illustration shows how such an process could look like. At first the backup gets triggered either by a human or another trigger (e.g. time-trigger). The operator instructs its watched resource (application) to set up a consistent state (like a consistent snapshot). Afterwards, the data of the application gets backed up using appropriate tools to an external storage. This could either be a one step process (backup directly to external storage) or in multiple steps, like writing to a persistent volume at first and to the external storage afterwards. The external storage might be a NFS/CIFS share (or any other network file system) on-premises, but also an object store/bucket on a cloud provider infrastructure. Either if the backup failed or succeeded, the state (of the backup) including the backed up application version and the location of the backup might be written to the state section of the custom resource.

Reviewer: ```suggestion
The above illustration shows how such a process could look like. At first the backup gets triggered either by a human or another trigger (e.g. time-trigger). The operator instructs its watched resource (application) to set up a consistent state (like a consistent snapshot). Afterwards, the data of the application gets backed up to an external storage using appropriate tools. This could either be a one-step process (backup directly to external storage) or in multiple steps, like writing to a persistent volume at first and to the external storage afterwards. The external storage might be a NFS/CIFS share (or any other network file system) on-premises, but also an object store/bucket on a cloud provider infrastructure. Whether the backup failed or succeeded, the state (of the backup) including the backed up application version and the location of the backup might be written to the state section of the custom resource.
```

#### Recovery from backup

> The recovery capability of an operator might assist a user to restore the application state from a successful backup. Therefore, the application state (application version and data) should be restored. 

Reviewer: ```suggestion
The recovery capability of an operator might assist a user in restoring the application state from a successful backup. Therefore, the application state (application version and data) should be restored. 
```
